### PR TITLE
Remove special Maven check when `-e` is used

### PR DESCRIPTION
**Pull request by erichaagdev**
### Maven

```
./02-validate-local-build-caching-different-locations.sh \
  -r https://github.com/apache/maven \
  -g package -x -e
```

image

---

```
./02-validate-local-build-caching-different-locations.sh \
  -r https://github.com/apache/maven \
  -g package -e
```

image

---

```
./02-validate-local-build-caching-different-locations.sh \
  -r https://github.com/apache/maven \
  -g package -e -s https://ge.solutions-team.gradle.com
```

image

### Gradle

```
./03-validate-local-build-caching-different-locations.sh \
  -r https://github.com/etiennestuder/java-ordered-properties \
  -t assemble -x -e
```

image

---

```
./03-validate-local-build-caching-different-locations.sh \
  -r https://github.com/etiennestuder/java-ordered-properties \
  -t assemble -e
```

image

---

```
./03-validate-local-build-caching-different-locations.sh \
  -r https://github.com/etiennestuder/java-ordered-properties \
  -t assemble -e -s https://ge.solutions-team.gradle.com
```

image

### CODE REVIEW

File: components/scripts/lib/config.sh [hunk 1]
```diff
@@ -84,10 +84,8 @@ validate_required_config() {
     fi
   fi
 
-  if [[ "${enable_ge}" == "on" && ( "${build_scan_publishing_mode}" == "on" || "${BUILD_TOOL}" = "Maven" ) ]]; then
-    if [ -z "${ge_server}" ]; then
-      _PRINT_HELP=yes die "ERROR: --gradle-enterprise-server is required when using --enable-gradle-enterprise." "${INVALID_INPUT}"
-    fi
+  if [[ "${enable_ge}" == "on" && -z "${ge_server}" && "${build_scan_publishing_mode}" == "on" ]]; then
```
Comment (erichaagdev) [inline components/scripts/lib/config.sh:87]: Note: I also removed the nested `if`.

Comment (etiennestuder) [reply components/scripts/lib/config.sh:87]: LGTM

**Review by etiennestuder — APPROVED** [review]

**Comment by etiennestuder** [comment]
@erichaagdev Before merging, please do a counter-test where publishing is enabled (and same on the Gradle side). If all is good, merge it. Thanks.

**Comment by erichaagdev** [comment]
@etiennestuder additional tests ran. Results have been added to the description.

Merging.

**Pull request merged**